types/redux-form/lib/Field.d.ts

```diff
@@ -17,7 +17,7 @@ export type Parser = (value: any, name: string) => any;
 export type Validator = (value: any, allValues?: any, props?: any, name?: any) => any;
 
 export type EventHandler<Event> = (event: Event) => void;
-export type EventWithDataHandler<Event> = (event?: Event, newValue?: any, previousValue?: any) => void;
+export type EventWithDataHandler<Event> = (event?: Event, newValue?: any, previousValue?: any, fieldName?: any) => void;
```

Comment:
As per official `redux-form` flow type definitions, the `fieldName` is of type `string`, beside being optional

+ I would recommend to name it `name` to keep it consistent with `redux-form`

```suggestion
export type EventWithDataHandler<Event> = (event?: Event, newValue?: any, previousValue?: any, name?: string) => void;
```